### DISCUSSION TITLE
Functional components?

### DISCUSSION THREAD

**Discussion by donovanhubbard** [vercel/hyper#7116] [General]
I'm new to react, electron, and hyper, and I've noticed that hyper plugins are based on higher order components and class based components. My understanding is that class based components are no longer recommended by the react authors (<https://react.dev/reference/react/Component>) and higher order components have fallen out of favor (<https://legacy.reactjs.org/docs/higher-order-components.html>) . I'm sure this discussion has been had many times before, but I couldn't find it. Does hyper support function based components for it's plugins or are there plans to support it in the future?

**Comment by LabhanshAgrawal** [comment]
yeah, we want to move slowly to functional components, you can still use functional components in plugins as they work without issue.
